### PR TITLE
Mat 5346 parsing errors

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-5346](https://jira.cms.gov/browse/MAT-5346)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/mat/cql_elm_translation/service/filters/CqlTranslatorExceptionFilter.java [hunk 1]
```diff
@@ -39,7 +40,17 @@ public List<CqlCompilerException> filter() {
       if (filteredCqlTranslatorExceptions.isEmpty()) {
         return Collections.emptyList();
       } else {
-        return filterByLibrary(filteredCqlTranslatorExceptions);
+        List<CqlCompilerException> filteredList = filterByLibrary(filteredCqlTranslatorExceptions);
+        List<CqlCompilerException> newList = new ArrayList<CqlCompilerException>();
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/filters/CqlTranslatorExceptionFilter.java]:
```suggestion
        List<CqlCompilerException> newList = new ArrayList<>();
```

Type declaration is no longer needed on the right hand side. Recommend using [diamond operator](https://www.baeldung.com/java-diamond-operator) instead.

File: src/main/java/gov/cms/mat/cql_elm_translation/service/filters/CqlTranslatorExceptionFilter.java [hunk 2]
```diff
@@ -39,7 +40,17 @@ public List<CqlCompilerException> filter() {
       if (filteredCqlTranslatorExceptions.isEmpty()) {
         return Collections.emptyList();
       } else {
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/filters/CqlTranslatorExceptionFilter.java]: `else` statement can be safely removed. The `return` statement will prevent the current `else` block from executing.

The `else` statement above on line 37 can also be removed.

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/filters/CqlTranslatorExceptionFilter.java]: @jkotanchik-SB Thanks a lot Joe! Just did the changes.

**Review by jkotanchik-SB — DISMISSED** [review]

**Comment by RohitKandimalla** [comment]
@CeciliaLiu8 Codacy Static Code Analysis identified a couple of changes. Looks like they can be addressed.

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**